core/services/workflows/syncer/workflow_registry.go

```diff
@@ -137,61 +187,116 @@ func WithReader(reader types.ContractReader) func(*workflowRegistry) {
 	}
 }
 
+// TODO Beholder implementation of this
+type workflowRegistryStats interface {
+	IncrementRegisteredWorkflowCount()
+	DecrementRegisteredWorkflowCount()
+}
+
 // NewWorkflowRegistry returns a new workflowRegistry.
 // Only queries for WorkflowRegistryForceUpdateSecretsRequestedV1 events.
 func NewWorkflowRegistry[T ContractReader](
 	lggr logger.Logger,
+	donID uint32,
```

Comment:
@ettec This probably won't work: the donID comes from the capabilities registry, so will need to be lazy loaded.

Reply:
I'm assuming the registry should not be used until its initial state has been loaded, and for this it needs the DON ID, so rather than have the registry check if it has been initialised yet on any method calls, it would probably be better to have the code that will handles registry instantiation take care of this IMO